src/ensembl/src/content/app/browser/drawer/drawerState.ts

```diff
@@ -25,10 +25,12 @@ export type DrawerState = Readonly<{
   isDrawerOpened: { [genomeId: string]: boolean };
   drawerView: { [genomeId: string]: DrawerView | null };
   activeDrawerTrackIds: { [genomeId: string]: string | null };
+  activeDrawerTranscriptIds: { [genomeId: string]: string | null };
 }>;
 
 export const defaultDrawerState = {
   isDrawerOpened: {},
   drawerView: {},
-  activeDrawerTrackIds: {}
+  activeDrawerTrackIds: {},
+  activeDrawerTranscriptIds: {}
```

Comment:
One suggestion that I offered during the code review was to see whether the drawer view name can be co-located with the data required by this view, e.g.:

```ts

type TranscriptViewType = {
  viewName: "transcriptView";
  payload: {
    transcriptId: "ENST1"
  }
}



type DrawerView = 
  | GeneViewType
  | TranscriptViewType

// and then the relevant piece of the state would look something like:
{
  drawerView: {
    "human": {
      viewName: "transcriptView",
      payload: {
        transcriptId: "ENST1"
      }
    }
  },
}
```

This may turn out to be a rather big change to make though. And it's hard to predict in advance whether  typescript is going to be upset about this.